core/src/main/javascript/app/pages/BackfillCreate.js

```diff
@@ -0,0 +1,223 @@
+// @flow
+
+import React from "react";
+import { connect } from "react-redux";
+import injectSheet from "react-jss";
+import goBack from "redux-url";
+import CloseIcon from "react-icons/lib/md/close";
+
+import Window from "../components/Window";
+import FancyTable from "../components/FancyTable";
+import JobSelector from "../components/JobSelector";
+import type { Workflow } from "../../datamodel";
+
+type Props = {
+  workflow: ?Workflow,
+  classes: any,
+  back: () => void
+};
+
+type State = {
+  selectedJobs: Array<string>
+};
+
+class BackfillCreate extends React.Component {
+  props: Props;
+  state: State;
+
+  constructor(props: Props) {
+    super(props);
+    (this: any).createBackfill = this.createBackfill.bind(this);
+    (this: any).selectJobs = this.selectJobs.bind(this);
+    this.state = {
+      selectedJobs: []
+    };
+  }
+
+  selectJobs(jobs: Array<string>) {
+    this.setState({ selectedJobs: jobs });
+  }
+
+  createBackfill(e) {
+    e.preventDefault();
+    const { name, start, end } = this.refs;
+    const { selectedJobs } = this.state;
+    return Promise.all(
+      selectedJobs.map(job =>
+        fetch(
+          `/api/timeseries/backfill?job=${job}&startDate=${start.value}&endDate=${end.value}&priority=0`,
+          { method: "POST" }
+        )
+      )
+    );
+  }
+
+  render() {
+    let { classes, workflow, back } = this.props;
+    let { selectedJobs } = this.state;
+
+    return (
+      <Window title="Create Backfill">
+        <CloseIcon className={classes.close} onClick={back} />
+        <form onSubmit={this.createBackfill}>
+          <FancyTable key="properties">
+            <dt key="name_">Name:</dt>
+            <dd key="name"><input ref="_name" type="text" required /></dd>
```

Comment:
I don't think you need to keep a reference to these form elements. The state of the main component should contain their data, and you can you onChange event handlers to handle state changes.